### PR TITLE
HTTP 2022: add queries for getting more preload stats and HTTP/3 support

### PR DESCRIPTION
**Pull request by paivaspol**
Add more stats for link preload hints and HTTP/3 support for #2903.

### CODE REVIEW

File: sql/2022/http/sites_using_link_preload_header.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      for (let linkTokenProp of linkTokenProps) {
+        if (linkTokenProps.includes('rel=preload')) {
+          numPreload++;
+        }
+      }
+    }
+  }
+  return numPreload;
+} catch (e) {
+  return [];
```
Comment (rviscomi) [inline sql/2022/http/sites_using_link_preload_header.sql:27]: The return value is numeric so this type should be comparable
```suggestion
  return 0;
```

File: sql/2022/http/sites_using_link_preload_header.sql [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#standardSQL
+
+# Number of <link rel="preload">.
+
+CREATE TEMPORARY FUNCTION getNumLinkRelPreloadRespHeader(HTTPheaders STRING, header STRING)
+RETURNS FLOAT64 LANGUAGE js AS """
```
Comment (rviscomi) [inline sql/2022/http/sites_using_link_preload_header.sql:6]:
```suggestion
RETURNS INT64 LANGUAGE js AS """
```

File: sql/2022/http/sites_using_link_preload_header.sql [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+FROM (
+  SELECT
+    client,
+    page,
+    getNumLinkRelPreloadRespHeader(response_headers, 'link') AS num_link_rel_preload_resp_header
+  FROM
+    `httparchive.almanac.requests`
+  WHERE
+    date = '2022-06-01' AND
+    firstHtml
```
Comment (rviscomi) [inline sql/2022/http/sites_using_link_preload_header.sql:45]: Need to group by page in order to count page-level stats in the outer query.
```suggestion
  SELECT
    client,
    page,
    SUM(getNumLinkRelPreloadRespHeader(response_headers, 'link')) AS num_link_rel_preload_resp_header
  FROM
    `httparchive.almanac.requests`
  WHERE
    date = '2022-06-01' AND
    firstHtml
  GROUP BY
    client,
    page
```

Comment (paivaspol) [reply sql/2022/http/sites_using_link_preload_header.sql]: done

File: sql/2022/http/sites_using_link_preload_header.sql (hunk 2 above)
Comment (paivaspol) [reply sql/2022/http/sites_using_link_preload_header.sql]: done

File: sql/2022/http/sites_using_link_preload_header.sql (hunk 1 above)
Comment (paivaspol) [reply sql/2022/http/sites_using_link_preload_header.sql]: done

Comment (rviscomi) [reply sql/2022/http/sites_using_link_preload_header.sql]: I don't think we want -1 since an error would undo 1 valid count

**Review by tunetheweb — APPROVED** [review]

**Pull request merged**